include/nbl/builtin/glsl/sampling/spherical_rectangle.glsl

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_GLSL_SAMPLING_SPHERICAL_RECTANGLE_INCLUDED_
+#define _NBL_BUILTIN_GLSL_SAMPLING_SPHERICAL_RECTANGLE_INCLUDED_
+
+#include <nbl/builtin/glsl/math/quaternions.glsl>
+#include <nbl/builtin/glsl/shapes/rectangle.glsl>
+
+#define CLAMP_EPS 1e-5f
+
+// Code from https://www.arnoldrenderer.com/research/egsr2013_spherical_rectangle.pdf
+vec3 nbl_glsl_sampling_generateSphericalRectangleSample(in vec3 r0, in vec3 r1, in float S, in float b0, float b1, in float k, vec2 uv)
+{
+    // 1. compute ’cu’
+    float au = uv.x * S + k;
```

Comment:
compute `cos_au` and `sin_au` from `nbl_glsl_sincos(uv.x*S)`, `cos_k` and `sin_k`, instead using angle sum trig indentifies

Or maybe `cot_au` and `sin_au` ?